manifests/config.pp

```diff
@@ -6,11 +6,17 @@
 class icingadb::config {
   assert_private()
 
+  $stdlib_version  = $icingadb::globals::stdlib_version
   $conf_dir        = $icingadb::globals::conf_dir
   $user            = $icingadb::globals::user
   $group           = $icingadb::globals::group
   $redis_tls_files = $icingadb::redis_tls_files
   $db_tls_files    = $icingadb::db_tls_files
+  $config_file     = if versioncmp($stdlib_version, '9.0.0') < 0 {
+    to_yaml($icingadb::config)
+  } else {
+    stdlib::to_yaml($icingadb::config)
+  }
```

Comment:
There was talk of introducing a stdlib 8.7.0 with the missing deprecated functions, but I don't think anything happened on that front.

I wonder: can you use `if defined("stdlib::to_yaml')` to dynamically detect it?

Reply:
If it should be that simple ...

Reply:
... unfortunately it is not.